packages/react/dropdown-menu/src/DropdownMenu.tsx

```diff
@@ -138,45 +174,54 @@ type DropdownMenuContentPrimitive = Polymorphic.ForwardRefComponent<
 
 const DropdownMenuContent = React.forwardRef((props, forwardedRef) => {
   const {
-    onCloseAutoFocus,
     disableOutsidePointerEvents = true,
     disableOutsideScroll = true,
     portalled = true,
     ...contentProps
   } = props;
   const context = useDropdownMenuContext(CONTENT_NAME);
+
+  const commonProps = {
+    ...contentProps,
+    disableOutsidePointerEvents,
+    disableOutsideScroll,
+    portalled,
+    style: {
+      ...contentProps.style,
+      // re-namespace exposed content custom property
+      ['--radix-dropdown-menu-content-transform-origin' as any]: 'var(--radix-popper-transform-origin)',
+    },
+  };
+
   return (
-    <SubmenuContext.Provider value={true}>
-      <MenuPrimitive.Content
-        id={context.contentId}
-        {...contentProps}
-        ref={forwardedRef}
-        disableOutsidePointerEvents={disableOutsidePointerEvents}
-        disableOutsideScroll={disableOutsideScroll}
-        portalled={portalled}
-        style={{
-          ...props.style,
-          // re-namespace exposed content custom property
-          ['--radix-dropdown-menu-content-transform-origin' as any]: 'var(--radix-popper-transform-origin)',
-        }}
-        trapFocus
-        onCloseAutoFocus={composeEventHandlers(onCloseAutoFocus, (event) => {
-          event.preventDefault();
-          context.triggerRef.current?.focus();
-        })}
-        onPointerDownOutside={composeEventHandlers(
-          props.onPointerDownOutside,
-          (event) => {
-            const target = event.target as HTMLElement;
-            const targetIsTrigger = context.triggerRef.current?.contains(target);
-            // prevent dismissing when clicking the trigger
-            // as it's already setup to close, otherwise it would close and immediately open.
-            if (targetIsTrigger) event.preventDefault();
-          },
-          { checkForDefaultPrevented: false }
-        )}
-      />
-    </SubmenuContext.Provider>
+    <ContentContext.Provider value={true}>
+      {context.isRootMenu ? (
+        <MenuPrimitive.Content
+          id={context.contentId}
```

Comment:
I would expect `id` to be a common prop no?

Reply:
It's not a common prop though. In `DropdownMenu` it's only needed to associate the root trigger with the root menu via `aria-controls`. 

The `Menu` handles the `aria-controls` stuff for `MenuSubTrigger` already and if I left put this `id` on both (when it isn't necessary) it would override the `id` that `Menu` applies from its context and break the `aria-controls` stuff in `Menu`.

Reply:
Ha right, that's done in `Menu`. This is doing my head in everytime, it's kinda hard to wrap your head around what's done where…

It's just that when I look at this code I'm like, well sumenus also need to associate their content to their triggers, so it looks like it's missing, but when you say it's done in `Menu` because it's shared, then it makes perfect sense.

Reply:
Perhaps we can add some comments for these cases to save our future selves from getting confused.

Reply:
I dunno if a comment is needed here honestly... `DropdownMenu` introduces a **new** `Trigger` part so it logically needs to add `aria-controls` stuff from that **new** part. It doesn't need to do that for submenus because it doesn't introduce a new dropdown specific part/relationship there, it just re-uses all the `Menu` parts.

For example, we don't do `aria-controls` stuff in `Accordion` for the `Collapsible` parts, we just re-compose it.

Reply:
Yeah that's very true, that's probably the diff that what caught my attention.